src/frontends/lean/definition_cmds.cpp

```diff
@@ -678,6 +680,7 @@ class proof_elaboration_task : public task<expr> {
             finalize_theorem_proof(elab, params, val, m_finfo);
             if (m_is_rfl_lemma && !is_rfl_lemma(m_final_type, val))
                 throw exception("not a rfl-lemma, even though marked as rfl");
+            if (elab.has_errors()) throw std::exception();
```

Comment:
The reason is the following workaround: in order not to forget any exceptions that occur during tasks, we catch exceptions that are subclasses of lean::throwable and report them automatically.  Throwing std::exception fails the task without producing an extra message.

So this is not related to the elaborator, but just to the tasks.